crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -841,7 +843,7 @@ impl<'a> Lexer<'a> {
             builder.push_matching(c);
         }
         self.read_decimal_digits(builder);
-        Kind::Float
+        if c == '-' { Kind::NegativeExponential } else { Kind::PositiveExponential }
```

Comment:
Nit: make the code cleaner (and less if checks) by combining this check with the `matches` above.